### PR TITLE
Evaluation module

### PR DESCRIPTION
**Pull request by mayanksinghrathore295**
his pull request includes the following changes:
1.	This is the Evaluation Dashboard in which assessment comes by assessmentid there are two status complete and pending,complete means evaluation us complete and pending means means evaluation is not yet completed now the common table component has been used for this dasboard
Screenshot 2024-12-11 140704
Screenshot 2024-12-11 140644
 
 
2.	When we click on the evaluate button the new route evaluate gets opened where all the questions are shown with their user responses the single answered and multi answered comes evaluated whereas in descriptive answers manual intervention is needed
 
Screenshot 2024-12-09 221530

3.	According to the answer the admin can give marks to the user but if the marks exceeds the assigned marks to that question a warning message is shown
 
Screenshot 2024-12-09 221627

4.	For single answered questions a radio box is shown with the answers given by the user which is disabled
 
Screenshot 2024-12-09 221720

5.	For multi answered questions check box are shown if the user answer is wrong it shown in red and if it is right it is hsown in green
 
Screenshot 2024-12-09 221750

 Screenshot 2024-12-11 140614
Screenshot 2024-12-09 222502

6.	After evaluating the descriptive questions the admin can make the user pass or fail and after making user pass or fail the admin can complete evaluation
 
Screenshot 2024-12-09 221825

7.	As soon as admin clicks on complete evaluation the view route gets opened and result is shown
 
Screenshot 2024-12-09 224023

8.	After that we can go back to the dashboard
 
Screenshot 2024-12-09 221825

### CODE REVIEW

**Review by sanjay611080 — DISMISSED** [review]

File: src/app/components/evaluation/service/evaluation-service.service.ts
Comment (goelp30) [inline src/app/components/evaluation/service/evaluation-service.service.ts:1]: do we really need this file, i can't see any code in this

File: src/app/components/evaluation/service/evaluation-service.service.spec.ts
Comment (goelp30) [inline src/app/components/evaluation/service/evaluation-service.service.spec.ts:1]: do we really need this file, i can't see any code in this

File: src/app/components/evaluation/service/evaluation.service.spec.ts
Comment (goelp30) [inline src/app/components/evaluation/service/evaluation.service.spec.ts:1]: you can delete this file

File: src/app/components/evaluation/view-assessment/view-assessment.component.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  templateUrl: './view-assessment.component.html',
+  styleUrl: './view-assessment.component.css'
+})
+export class ViewAssessmentComponent {
+
+ clickedData:any;
+  constructor(private evaluationService: EvaluationService,private router:Router){}
+  ngOnInit(): void {
+  this.clickedData = this.evaluationService.getData();
+  console.log(this.clickedData);
```
Comment (goelp30) [inline src/app/components/evaluation/view-assessment/view-assessment.component.ts:20]: remove console

File: src/app/components/evaluation/view-assessment/view-assessment.component.ts
Comment (goelp30) [inline src/app/components/evaluation/view-assessment/view-assessment.component.ts:1]: add comments which function doing what

**Comment by goelp30** [comment]
First update your branch with develop

File: src/app/components/evaluation/commonEvaluationComponent/marks/marks.component.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  standalone: true,
+  imports: [],
+  templateUrl: './marks.component.html',
+  styleUrl: './marks.component.css'
+})
+export class MarksComponent {
+  AssessementName: string = "Angular"
+  obtainMark:number = 50;
+  TotalMark: number = 49;
+
```
Comment (goelp30) [inline src/app/components/evaluation/commonEvaluationComponent/marks/marks.component.ts:14]: why white space is left

File: src/app/components/evaluation/commonEvaluationComponent/questions/questions.component.ts
Comment (goelp30) [inline src/app/components/evaluation/commonEvaluationComponent/questions/questions.component.ts:1]: what this component is doing , why there is mock data in this

File: src/app/components/evaluation/service/evaluation-service.service.ts
Comment (mayanksinghrathore295) [reply src/app/components/evaluation/service/evaluation-service.service.ts:1]: I have deleted that file now

File: src/app/components/evaluation/service/evaluation-service.service.spec.ts
Comment (mayanksinghrathore295) [reply src/app/components/evaluation/service/evaluation-service.service.spec.ts:1]: Deleted

File: src/app/components/evaluation/service/evaluation.service.spec.ts
Comment (mayanksinghrathore295) [reply src/app/components/evaluation/service/evaluation.service.spec.ts:1]: Deleted

File: src/app/components/evaluation/view-assessment/view-assessment.component.ts (hunk 1 above)
Comment (mayanksinghrathore295) [reply src/app/components/evaluation/view-assessment/view-assessment.component.ts]: Removed

File: src/app/components/evaluation/view-assessment/view-assessment.component.ts
Comment (mayanksinghrathore295) [reply src/app/components/evaluation/view-assessment/view-assessment.component.ts:1]: added

**Review by kasharora03 — APPROVED** [review]
approve

**Pull request merged**